autopush/utils.py

```diff
@@ -108,13 +108,38 @@ def fix_padding(string):
     return string
 
 
+def decipher_public_key(key_data):
+    """A public key may come in several flavors. Attempt to extract the
+    valid key bits from keys doing minimal validation checks.
+
+    This is mostly a result of libs like WebCrypto prefixing data to "raw"
+    keys, and the ecdsa library not really providing helpful errors.
+
+    :param key_data: the raw-ish key we're going to try and process
+    :returns: the raw key data.
+    :raises: ValueError for unknown or poorly formatted keys.
+
+    """
+    # key data is actually a raw coordinate pair
+    if len(key_data) == 64:
+        return key_data
+    # Key format is "raw"
+    if len(key_data) == 65 and key_data[0] == '\x04':
+        return key_data[-64:]
+    # key format is "spki"
+    if len(key_data) == 88 and key_data[:3] == '0V0':
+        return key_data[-64:]
```

Comment:
Do the len call just once and store the result to avoid unnecessary repeat func calls.